core/common/src/main/java/alluxio/exception/status/AlluxioStatusException.java

```diff
@@ -189,6 +189,8 @@ public static AlluxioStatusException from(Status status, String m) {
   public static AlluxioStatusException fromCheckedException(Throwable throwable) {
     try {
       throw throwable;
+    } catch (AlluxioStatusException e) {
```

Comment:
this case should already be handled because `AlluxioStatusException` extends `IOException`

Reply:
You are right it is not necessary. This change does not actually fix the issue that we are returning status code 500 for a temporary unavailable state. I will try to fix it in a separate PR.